src/locales/it/parsers/ITMonthNameMiddleEndianParser.ts

```diff
@@ -0,0 +1,87 @@
+import { ParsingContext } from "../../../chrono";
+import { findYearClosestToRef } from "../../../calculation/years";
+import { MONTH_DICTIONARY } from "../constants";
+import { ORDINAL_NUMBER_PATTERN, parseOrdinalNumberPattern } from "../constants";
+import { YEAR_PATTERN, parseYear } from "../constants";
+import { matchAnyPattern } from "../../../utils/pattern";
+import { AbstractParserWithWordBoundaryChecking } from "../../../common/parsers/AbstractParserWithWordBoundary";
+
+const PATTERN = new RegExp(
+    "(?:il\\s*?|dal\\.*)?" +
+        `(${ORDINAL_NUMBER_PATTERN})` +
+        "(?:\\s*" +
+        "(?:tot|\\-|\\–|\\w*?al\\.*|\\w*?ino\\s*a\\.*?|\\s)\\s*" +
+        `(${ORDINAL_NUMBER_PATTERN})` +
+        ")?" +
+        "(?:-|/|\\s*(?:di)?\\s*)" +
+        "(" +
+        matchAnyPattern(MONTH_DICTIONARY) +
+        ")" +
+        "(?:" +
+        "(?:-|/|,?\\s*)" +
+        `(${YEAR_PATTERN}(?![^\\s]\\d))` +
+        ")?" +
+        "(?=\\W|$)",
+    "i"
+);
+
+const MONTH_NAME_GROUP = 3;
+const DATE_GROUP = 1;
+const DATE_TO_GROUP = 2;
+const YEAR_GROUP = 4;
+
+/**
+ * The parser for parsing IT date format with month's name in full writing
+ *  - 1° gennaio 2019
+ *  - 01 gennaio 2019
+ *  - 10 gennaio 2019
+ *  - 13 gennaio
+ *  - 10-25 marzo
+ *  - 10-25 marzo 2019
+ *  - 1° ago 2019
+ *  - 1° settembre 200 a.C.
+ *  - 1° settembre 2002 d.C.
+ *  - 19 gennaio 87
+ *  - 12 giugno 2013
+ *  - 1° novembre 2013
+ */
+
+export default class ITMonthNameMiddleEndianParser extends AbstractParserWithWordBoundaryChecking {
```

Comment:
From your example and test, this is Little-endian (i.e., day-month-year), not Middle-endian.